File: docs-old/notintoc/codesnippet/CSharp/walkthrough--customizing-toolbox-item-configuration-dynamically_5.cs
        [PrincipalPermission(SecurityAction.Demand)]
        public void ConfigureToolboxItem(ToolboxItem item)
        {
            if (null == item) return;

            // Create a filter for the Toolbox.
            ToolboxItemFilterAttribute newFilter;
            if (typeof(Control1).ToString() == item.TypeName)
            {
                // For Control1, only show it when editing a UserControl.
                newFilter = new ToolboxItemFilterAttribute(
                    "System.Windows.Forms.UserControl",
                    ToolboxItemFilterType.Require);
            }
            else if (typeof(Control2).ToString() == item.TypeName)
            {
                // For Control2, only show it when not editing a UserControl.
                newFilter = new ToolboxItemFilterAttribute(
                    "System.Windows.Forms.UserControl",
                    ToolboxItemFilterType.Prevent);
            }
            else
            {
                // Don't apply a filter to other classes.
                return;
            }

            item.Filter = (ICollection)(new ToolboxItemFilterAttribute[] { newFilter });
        }

File: docs/misc/codesnippet/VisualBasic/walkthrough-customizing-toolbox-item-configuration-dynamically_5.vb
    <PrincipalPermission(SecurityAction.Demand)> Public Sub _
        ConfigureToolboxItem(ByVal item As ToolboxItem) _
        Implements IConfigureToolboxItem.ConfigureToolboxItem

        If item Is Nothing Then Return

        ' Create a filter for the Toolbox.
        Dim newFilter As ToolboxItemFilterAttribute
        If GetType(Control1).ToString() = item.TypeName Then

            ' For Control1, only show it when editing a UserControl.
            newFilter = New ToolboxItemFilterAttribute( _
                "System.Windows.Forms.UserControl", _
                ToolboxItemFilterType.Require)

        ElseIf GetType(Control2).ToString() = item.TypeName Then

            ' For Control2, only show it when not editing a UserControl.
            newFilter = New ToolboxItemFilterAttribute( _
                "System.Windows.Forms.UserControl", _
                ToolboxItemFilterType.Prevent)

        Else

            ' Don't apply a filter to other classes.
            Return

        End If

        item.Filter = CType(New ToolboxItemFilterAttribute() {newFilter}, ICollection)
    End Sub